server/src/CompilationDatabase.cpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * \file
+ *
+ * \brief Using libclang for reading the JSON compilation database.
+ *
+ * This file is distributed under the GNU General Public License. See
+ * COPYING for details.
+ */
+#include <iostream>
+
+#include <clang-c/Index.h>
+#include <clang-c/CXCompilationDatabase.h>
+#include <clang-c/CXString.h>
+
+#include "CompilationDatabase.h"
+
+static void getExactFlags(std::vector<std::vector<std::string>> &flags,
```

Comment:
Out parameters goes last. But here it shouldn't be so much of a performance bottleneck to return by value, I think it will be more natural to do so.